src/main/scala/io/iohk/ethereum/consensus/pow/miners/EthashMiner.scala

```diff
@@ -0,0 +1,136 @@
+package io.iohk.ethereum.consensus.pow.miners
+
+import akka.actor.{Actor, ActorLogging, Props, ActorRef => ClassicActorRef}
+import akka.util.ByteString
+import io.iohk.ethereum.blockchain.sync.SyncProtocol
+import io.iohk.ethereum.consensus.blocks.{PendingBlock, PendingBlockAndState}
+import io.iohk.ethereum.consensus.pow.PoWMiningCoordinator.CoordinatorProtocol
+import io.iohk.ethereum.consensus.pow.miners.MinerProtocol._
+import io.iohk.ethereum.consensus.pow.{EthashUtils, PoWBlockCreator, PoWMiningCoordinator}
+import io.iohk.ethereum.crypto
+import io.iohk.ethereum.domain.{Block, BlockHeader}
+import io.iohk.ethereum.jsonrpc.EthMiningService
+import io.iohk.ethereum.jsonrpc.EthMiningService.SubmitHashRateRequest
+import io.iohk.ethereum.utils.BigIntExtensionMethods._
+import io.iohk.ethereum.utils.{ByteStringUtils, ByteUtils}
+import monix.execution.Scheduler
+
+import scala.util.Random
+
+/**
+  * Implementation of Ethash CPU mining worker.
+  * Could be started by switching configuration flag "consensus.mining-enabled" to true
+  * Implementation explanation at https://eth.wiki/concepts/ethash/ethash
+  */
+class EthashMiner(
+    dagManager: EthashDAGManager,
+    blockCreator: PoWBlockCreator,
+    syncController: ClassicActorRef,
+    ethMiningService: EthMiningService
+) extends Actor
+    with ActorLogging {
+
+  import EthashMiner._
+
+  private implicit val scheduler: Scheduler = Scheduler(context.dispatcher)
+
+  override def receive: Receive = started
+
+  def started: Receive = { case ProcessMining(bestBlock, coordinatorRef) =>
+    log.debug("Received message ProcessMining with parent block {}", bestBlock.number)
+    processMining(bestBlock, coordinatorRef)
+  }
+
+  private def processMining(bestBlock: Block, coordinatorRef: akka.actor.typed.ActorRef[CoordinatorProtocol]): Unit = {
+    blockCreator
```

Comment:
since this is a one time `Task`, what's the need for an actor?

Reply:
The EthashMiner and KeccakMiner don't need to run any state. So indeed a Task seems to be enough. I will do that one later, as the tests also need to be adjusted

Reply:
I have refactored. Please check